From cd7f1c94536481a04190f590c8e5e323f6150659 Mon Sep 17 00:00:00 2001
From: "Yang, Dong" <dong.yang@intel.com>
Date: Wed, 20 May 2015 15:33:56 +0800
Subject: [PATCH 405/462] gc2235, set the mipi parameter according 120M

the mipi parameter need adjust according to mipi clock,
so far it's 120M for T3 platform, reconfig the mipi
parameter according this clock speed, and revert the
ISP default settle.

Change-Id: I93e7cd568592a813d43fd86d7623a3f97934e5f7
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-36926
Signed-off-by: Yang, Dong <dong.yang@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/56753
Reviewed-by: Tian Shu Qiu <tian.shu.qiu@intel.com>
Build: IMIN (SC)
Validation-Android: Wei Tao <wei.tao@intel.com>
Approver: Tian Shu Qiu <tian.shu.qiu@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../camera/drivers/media/i2c/gc2235.h              | 83 ++++++++++++++--------
 .../pci/atomisp2/atomisp_driver/atomisp_csi2.c     |  2 +-
 2 files changed, 53 insertions(+), 32 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h b/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h
index 66e9b25..ad80db8 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h
@@ -333,14 +333,20 @@ static struct gc2235_reg const gc2235_1296_736_30fps[] = {
 	{ GC2235_8BIT, 0x13, 0x06 }, /* val_high = (width * 10 / 8) >> 8 */
 
 	{ GC2235_8BIT, 0x15, 0x12 }, /* DPHY mode*/
-	{ GC2235_8BIT, 0x04, 0x20 },
+	{ GC2235_8BIT, 0x04, 0x10 },
 	{ GC2235_8BIT, 0x05, 0x00 },
 	{ GC2235_8BIT, 0x17, 0x01 },
-	{ GC2235_8BIT, 0x21, 0x01 },
-	{ GC2235_8BIT, 0x22, 0x02 },
-	{ GC2235_8BIT, 0x23, 0x01 },
-	{ GC2235_8BIT, 0x29, 0x02 },
-	{ GC2235_8BIT, 0x2a, 0x09 },
+
+	{ GC2235_8BIT, 0x22, 0x01 },
+	{ GC2235_8BIT, 0x23, 0x05 },
+	{ GC2235_8BIT, 0x24, 0x10 },
+	{ GC2235_8BIT, 0x25, 0x10 },
+	{ GC2235_8BIT, 0x26, 0x02 },
+	{ GC2235_8BIT, 0x21, 0x10 },
+	{ GC2235_8BIT, 0x29, 0x01 },
+	{ GC2235_8BIT, 0x2a, 0x02 },
+	{ GC2235_8BIT, 0x2b, 0x02 },
+
 	{ GC2235_8BIT, 0x10, 0x01 }, /* disable mipi */
 	{ GC2235_8BIT, 0xfe, 0x00 }, /* switch to P0 */
 	{ GC2235_TOK_TERM, 0, 0 }
@@ -380,14 +386,18 @@ static struct gc2235_reg const gc2235_960_640_30fps[] = {
 	{ GC2235_8BIT, 0x13, 0x04 }, /* val_high = (width * 10 / 8) >> 8 */
 
 	{ GC2235_8BIT, 0x15, 0x12 }, /* DPHY mode*/
-	{ GC2235_8BIT, 0x04, 0x20 },
+	{ GC2235_8BIT, 0x04, 0x10 },
 	{ GC2235_8BIT, 0x05, 0x00 },
 	{ GC2235_8BIT, 0x17, 0x01 },
-	{ GC2235_8BIT, 0x21, 0x01 },
-	{ GC2235_8BIT, 0x22, 0x02 },
-	{ GC2235_8BIT, 0x23, 0x01 },
-	{ GC2235_8BIT, 0x29, 0x02 },
-	{ GC2235_8BIT, 0x2a, 0x09 },
+	{ GC2235_8BIT, 0x22, 0x01 },
+	{ GC2235_8BIT, 0x23, 0x05 },
+	{ GC2235_8BIT, 0x24, 0x10 },
+	{ GC2235_8BIT, 0x25, 0x10 },
+	{ GC2235_8BIT, 0x26, 0x02 },
+	{ GC2235_8BIT, 0x21, 0x10 },
+	{ GC2235_8BIT, 0x29, 0x01 },
+	{ GC2235_8BIT, 0x2a, 0x02 },
+	{ GC2235_8BIT, 0x2b, 0x02 },
 	{ GC2235_8BIT, 0x10, 0x01 }, /* disable mipi */
 	{ GC2235_8BIT, 0xfe, 0x00 }, /* switch to P0 */
 	{ GC2235_TOK_TERM, 0, 0 }
@@ -421,14 +431,18 @@ static struct gc2235_reg const gc2235_1600_900_30fps[] = {
 	{ GC2235_8BIT, 0x13, 0x07 }, /* val_high = (width * 10 / 8) >> 8 */
 
 	{ GC2235_8BIT, 0x15, 0x12 }, /* DPHY mode*/
-	{ GC2235_8BIT, 0x04, 0x20 },
+	{ GC2235_8BIT, 0x04, 0x10 },
 	{ GC2235_8BIT, 0x05, 0x00 },
 	{ GC2235_8BIT, 0x17, 0x01 },
-	{ GC2235_8BIT, 0x21, 0x01 },
-	{ GC2235_8BIT, 0x22, 0x02 },
-	{ GC2235_8BIT, 0x23, 0x01 },
-	{ GC2235_8BIT, 0x29, 0x02 },
-	{ GC2235_8BIT, 0x2a, 0x09 },
+	{ GC2235_8BIT, 0x22, 0x01 },
+	{ GC2235_8BIT, 0x23, 0x05 },
+	{ GC2235_8BIT, 0x24, 0x10 },
+	{ GC2235_8BIT, 0x25, 0x10 },
+	{ GC2235_8BIT, 0x26, 0x02 },
+	{ GC2235_8BIT, 0x21, 0x10 },
+	{ GC2235_8BIT, 0x29, 0x01 },
+	{ GC2235_8BIT, 0x2a, 0x02 },
+	{ GC2235_8BIT, 0x2b, 0x02 },
 	{ GC2235_8BIT, 0x10, 0x01 }, /* disable mipi */
 	{ GC2235_8BIT, 0xfe, 0x00 }, /* switch to P0 */
 	{ GC2235_TOK_TERM, 0, 0 }
@@ -462,14 +476,18 @@ static struct gc2235_reg const gc2235_1616_1082_30fps[] = {
 	{ GC2235_8BIT, 0x13, 0x07 }, /* val_high = (width * 10 / 8) >> 8 */
 
 	{ GC2235_8BIT, 0x15, 0x12 }, /* DPHY mode*/
-	{ GC2235_8BIT, 0x04, 0x20 },
+	{ GC2235_8BIT, 0x04, 0x10 },
 	{ GC2235_8BIT, 0x05, 0x00 },
 	{ GC2235_8BIT, 0x17, 0x01 },
-	{ GC2235_8BIT, 0x21, 0x01 },
-	{ GC2235_8BIT, 0x22, 0x02 },
-	{ GC2235_8BIT, 0x23, 0x01 },
-	{ GC2235_8BIT, 0x29, 0x02 },
-	{ GC2235_8BIT, 0x2a, 0x01 },
+	{ GC2235_8BIT, 0x22, 0x01 },
+	{ GC2235_8BIT, 0x23, 0x05 },
+	{ GC2235_8BIT, 0x24, 0x10 },
+	{ GC2235_8BIT, 0x25, 0x10 },
+	{ GC2235_8BIT, 0x26, 0x02 },
+	{ GC2235_8BIT, 0x21, 0x10 },
+	{ GC2235_8BIT, 0x29, 0x01 },
+	{ GC2235_8BIT, 0x2a, 0x02 },
+	{ GC2235_8BIT, 0x2b, 0x02 },
 	{ GC2235_8BIT, 0x10, 0x01 }, /* disable mipi */
 	{ GC2235_8BIT, 0xfe, 0x00 }, /* switch to P0 */
 	{ GC2235_TOK_TERM, 0, 0 }
@@ -501,16 +519,19 @@ static struct gc2235_reg const gc2235_1616_1216_30fps[] = {
 	/* set mipi buffer */
 	{ GC2235_8BIT, 0x12, 0xe4 }, /* val_low = (width * 10 / 8) & 0xFF */
 	{ GC2235_8BIT, 0x13, 0x07 }, /* val_high = (width * 10 / 8) >> 8 */
-
 	{ GC2235_8BIT, 0x15, 0x12 }, /* DPHY mode*/
-	{ GC2235_8BIT, 0x04, 0x20 },
+	{ GC2235_8BIT, 0x04, 0x10 },
 	{ GC2235_8BIT, 0x05, 0x00 },
 	{ GC2235_8BIT, 0x17, 0x01 },
-	{ GC2235_8BIT, 0x21, 0x01 },
-	{ GC2235_8BIT, 0x22, 0x02 },
-	{ GC2235_8BIT, 0x23, 0x01 },
-	{ GC2235_8BIT, 0x29, 0x02 },
-	{ GC2235_8BIT, 0x2a, 0x01 },
+	{ GC2235_8BIT, 0x22, 0x01 },
+	{ GC2235_8BIT, 0x23, 0x05 },
+	{ GC2235_8BIT, 0x24, 0x10 },
+	{ GC2235_8BIT, 0x25, 0x10 },
+	{ GC2235_8BIT, 0x26, 0x02 },
+	{ GC2235_8BIT, 0x21, 0x10 },
+	{ GC2235_8BIT, 0x29, 0x01 },
+	{ GC2235_8BIT, 0x2a, 0x02 },
+	{ GC2235_8BIT, 0x2b, 0x02 },
 	{ GC2235_8BIT, 0x10, 0x01 }, /* disable mipi */
 	{ GC2235_8BIT, 0xfe, 0x00 }, /* switch to P0 */
 	{ GC2235_TOK_TERM, 0, 0 }
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_csi2.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_csi2.c
index 574fbe3..fd563ef 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_csi2.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_csi2.c
@@ -337,7 +337,7 @@ static void atomisp_csi2_configure_isp2401(struct atomisp_sub_device *asd)
 	static const short int coeff_dat_termen[] = { 0, 0 };
 	static const short int coeff_dat_settle[] = { 85, -2 };
 	static const int TERMEN_DEFAULT		  = 0 * 0;
-	static const int SETTLE_DEFAULT		  = 0x640;
+	static const int SETTLE_DEFAULT		  = 0x480;
 	static const hrt_address csi2_port_base[] = {
 		[ATOMISP_CAMERA_PORT_PRIMARY]     = CSI2_PORT_A_BASE,
 		[ATOMISP_CAMERA_PORT_SECONDARY]   = CSI2_PORT_B_BASE,
-- 
1.9.1

